patch_art.sh

```diff
@@ -3,34 +3,105 @@
 set -euo pipefail
 
 this_dir="$(cd "$(dirname "$0")" && pwd)"
-widgets_dir="${this_dir}/zmk/app/boards/shields/nice_view/widgets"
-ART_FILE="${widgets_dir}/art.c"
-PERIPHERAL_STATUS_FILE="${widgets_dir}/peripheral_status.c"
+art_dir="${this_dir}/art"
+config_json="${art_dir}/config.json"
 
-IMG_SIZE="68x140"
-
-reset_zmk() {
-    cd "${this_dir}/zmk"
+magick_args() {
+    local name="$1"
+
+    local config file resize color_convert magick_args
+    config="$(jq -c '.images[] | select(.name == "'"${name}"'") | {name,file,resize,color_convert,magick_args}' "${config_json}")"
+    file="$(         jq -r '.file          // false'  <<<"$config")"
+    resize="$(       jq -r '.resize        // false'  <<<"$config")"
+    color_convert="$(jq -r '.color_convert // "none"' <<<"$config")"
+    magick_args="$(  jq -r '.magick_args   // "-"'    <<<"$config")"
+
+    local src_file="${art_dir}/${file}"
+    local -a args=("${src_file}")
+
+    local IMG_SIZE="68x140"
+    case "$resize" in
+        proportional) # Image is expected to be proportional to 68x140
+            local size_modulus
+            size_modulus=$(file "${src_file}" | perl -p -e 's/.*?\s(\d+)\sx\s(\d+).*/($1\*140)%($2*68)/g' | bc)
+
+            if [[ "$size_modulus" != "0" ]]; then
+                local size
+                size=$(file "${src_file}" | perl -p -e 's/.*?\s(\d+\sx\s\d+).*/$1/g')
+                >&2 echo "ERROR: Image size \(${size}\) not proportional to 68x140. Use either \"ignoreAspectRatio\", or \"fill\" resize strategy instead."
+                exit 1
+            fi
+
+            args+=("-resize" "${IMG_SIZE}")
+            ;;
+        ignoreAspectRatio) # Image will be distorted
+            args+=("-resize" "${IMG_SIZE}!")
+            ;;
+        fill) # Resize the image based on the smallest fitting dimension 
+            args+=("-resize" "${IMG_SIZE}^" "-gravity" "center" "-extent" "${IMG_SIZE}")
+            ;;
+        *)
+            local size
+            size="$(file "${src_file}" | perl -p -e 's/.*?\s(\d+\sx\s\d+).*/$1/g')"
+            if [[ "$size" != "68 x 140" ]]; then
+                >&2 cat <<EOF
+ERROR: Invalid image size: "${size}" for $file. Use "resize" to set a resize strategy.
+Available resize strategies: "proportional", "ignoreAspectRatio", "fill"
+EOF
+                exit 1
+            fi
+            ;;
+    esac
+
+    case "$color_convert" in
+        "monochrome") args+=("-monochrome") ;;
+        "threshold")  args+=("-threshold" "50%") ;;
+        "h4x4a")      args+=("-colorspace" "Gray" "-ordered-dither" "h4x4a") ;;
+        *) >&2 echo "ERROR: Unsupported color_convert option \"${color_convert}\"" && exit 1;;
```

Comment:
```suggestion
        "h4x4a")      args+=("-colorspace" "Gray" "-ordered-dither" "h4x4a") ;;
        "none")       ;;
        *) >&2 echo "ERROR: Unsupported color_convert option \"${color_convert}\"" && exit 1;;
```

`none` (defaullt/blank value) needs handling. IDK if this is the best way.